server/src/main/java/org/elasticsearch/index/mapper/MetadataFieldMapper.java

```diff
@@ -173,4 +173,6 @@ public void postParse(DocumentParserContext context) throws IOException {
         // do nothing
     }
 
+    @Override
+    public abstract SourceLoader.SyntheticFieldLoader syntheticFieldLoader();
```

Comment:
I would like to make this abstract across the board just so everyone who makes a field has to think about it. If they don't want to support it, fine, but I'd like folks to think about it.

Reply:
But I didn't want to make it abstract in `Mapper` yet. It's just twice as many changes....